### PR TITLE
[TIMOB-20272] Attempt to fix notifications in cold app-start

### PR DESCRIPTION
**Pull request by hansemannn**
**JIRA:** https://jira.appcelerator.org/browse/TIMOB-20272

### CODE REVIEW

File: iphone/Classes/TiApp.m [hunk 1]
```diff
@@ -100,12 +100,19 @@ -(TiContextGroupRef)contextGroup
...
-
 +(TiContextGroupRef)contextGroup
 {
 	return [sharedApp contextGroup];
 }
 
+- (NSMutableDictionary *)queuedBootEvents
+{
+    if (queuedBootEvents == nil) {
+        queuedBootEvents = [[[NSMutableDictionary alloc] init] retain];
```
Comment (vijaysingh-axway) [inline iphone/Classes/TiApp.m]: Double retain. Is that required ?

Comment (hansemannn) [inline iphone/Classes/TiApp.m]: Retaining it to ensure it doesn't get GC'd during other lifetime events. I noticed that this sometimes happens and sometimes not, so I ensure to retain it here. Can you run the analyser to see if it's fine with it? Thanks!

**Review by vijaysingh-axway — APPROVED** [review]
CR passed.

**Comment by ludolphus** [comment]
You forgot to change lines 512 and 726. There postNotificationwithKey is still used instead of the new tryToPostBackgroundModeNotification. It crashed my app on receiving a remote push.

<Notice>: [ERROR] The application has crashed with an uncaught exception 'NSInvalidArgumentException'.
Reason:
-[TiApp postNotificationwithKey:withNotificationName:]: unrecognized selector sent to instance

Changing both to tryToPostBackgroundModeNotification fixes this.

**Comment by hansemannn** [comment]
@ludolphus Thanks for the eagle eyes! Just pushed a fix, @vijaysingh-axway can you validate the latest commit as well?

**Review by vijaysingh-axway — APPROVED** [review]
CR passed.

**Review by mukherjee2 — APPROVED** [review]
Passed FR using:
Node Version: 6.10.3
NPM Version: 3.10.10
Mac OS: 10.12.4
Appc CLI: 6.2.3
Appc CLI NPM: 4.2.9
Titanium SDK version: 6.2.0.v20170809123942
Appcelerator Studio, build: 4.9.1.201707200100
Xcode 8.3.2

Notifications were received while app was not in memory.

**Pull request merged**